### PR TITLE
Added sort forms and queries onto filterList.php

### LINKED ISSUE #61: Sort tasks in a task list
• Generate form for user to indicate sort conditions
• Write queries to retrieve sorted tasks based on conditions
• Conditions include:
- Task Title
- Importance
- Due Date
- Work type

### PR DESCRIPTION
**Pull request by AaronDzf**
- Users can sort tasks in a list by task title, importance, work type, due date via drop-down list of options
- Testing:
        - Sort filter will not apply if none selected

Changes to filterList.php:
- Restricted input of filters:
	-Title and type of work max length of 255 characters
	-Importance can only take integer values between 0 and 10
- Added importance, type of work, due date into text output of tasks

### CODE REVIEW

File: filterList.php [hunk 1]
```diff
@@ -47,28 +76,51 @@
         foreach($filtered_tasks as $task){
             echo "<br>Title:".$task['taskTitle']."<br>";
             echo "Description: ".$task['description']."<br>";
-    
+            echo "Importance: ".$task['importance']."<br>";
+            echo "Type of Work: ".$task['typeOfWork']."<br>";
+            echo "Due Date: ".gmdate("Y-m-d", $task['dueDate'])."<br>";
```
Comment (Rolo123y) [inline filterList.php:81]: This is a good addition! 👍

File: filterList.php [hunk 2]
```diff
@@ -47,28 +76,51 @@
...
 
     <br>
-    <input type="text" name="filt_name" value="" class="form-control" placeholder="Task Title"/>
+    <input type="text" name="filt_name" value="" class="form-control" placeholder="Task Title" maxlength="255"/>
     <br>
-    <input type="number" name="filt_effort" value="" class="form-control" placeholder="Importance"/>
+    <input type="number" name="filt_effort" value="" class="form-control" placeholder="Importance" min = "0" max = "10" step="1"/>
     <br>
-    <input type="text" name="filt_work" value="" class="form-control" placeholder="Type of Work"/>
+    <input type="text" name="filt_work" value="" class="form-control" placeholder="Type of Work" maxlength="255"/>
```
Comment (Rolo123y) [inline filterList.php:99]: good addition to the inputs. Users should be constrained so there are no errors when making a query! 👍

File: filterList.php [hunk 3]
```diff
@@ -47,28 +76,51 @@
...
     <br>
     <input type="datetime-local" name="filt_date" value="unchanged" class="form-control" placeholder="Due Date"/>
     <br>
+    <select name="sortTask">
+        <option value="" disabled selected>Sort Task By</option>
+        <optgroup label="Sort by Title">
+            <option value="ascTitle">Ascending</option>
+            <option value="descTitle">Descending</option>
+        </optgroup>
+        <optgroup label="Sort by important">
```
Comment (Rolo123y) [inline filterList.php]: This should say 'Sort by importance'

File: filterList.php [hunk 4]
```diff
@@ -47,28 +76,51 @@
...
+            <option value="ascEffort">Ascending</option>
+            <option value="descEffort">Descending</option>
+        </optgroup>
+        <optgroup label="Sort by work type">
+            <option value="ascType">Ascending</option>
+            <option value="descType">Descending</option>
+        </optgroup>
+        <optgroup label="Sort by date">
```
Comment (Rolo123y) [inline filterList.php]: This should be 'Sort by due-date'

**Review by Rolo123y — COMMENTED** [review]
Tested each individual sort with the same list:
1. Sort by Title
Ascending: sorts the titles correctly in ascending alphabetical order
Descending: sorts the titles correctly in descending alphabetical order 

2. sort by important
Ascending: sorts the importance of tasks correctly in ascending order
Descending: sorts the importance of tasks correctly in descending order

3. sort by work type
Ascending: sorts the work type of tasks correctly in ascending order
Descending: sorts the work type of tasks correctly in descending order

4. sort by date
Ascending: sorts the due-date of tasks correctly in ascending order
Descending: sorts the due-date of tasks correctly in descending order

File: filterList.php (hunk 3 above)
Comment (AaronDzf) [reply filterList.php]: Will make changes, thanks for the catch!

File: filterList.php (hunk 4 above)
Comment (AaronDzf) [reply filterList.php]: Will make changes, thanks for the catch!

**Review by Rolo123y — APPROVED** [review]

**Comment by Rolo123y** [comment]
Looks good to merge now!

**Comment by gaurav613** [comment]
This is a great modification to the filtering functionality! Thank you for adding the input restrictions to my code(I was going to do so after the changes were made to the 'add task' functionality). 😄

**Pull request merged**